privacyidea/lib/policymatch.py

```diff
@@ -0,0 +1,175 @@
+                                                                    unique=unique,
+                                                                    allow_white_space_in_action=
+                                                                    allow_white_space_in_action)
+        if write_to_audit_log:
+            for action_value, policy_names in action_values.items():
+                for p_name in policy_names:
+                    self._g.audit_object.audit_data.setdefault("policies", []).append(p_name)
+        return action_values
+
+    @classmethod
+    def simple(cls, g, scope, action, realm, user):
+        """
+        Simple policy matching with a scope, an action and optionally either a realm or a user.
+        :param g: ``flask.g`` object
+        :param scope: the policy scope. SCOPE.ADMIN cannot be passed, ``admin`` must be used instead.
+        :param action: the policy action
+        :param realm: the realm against which policies should be matched. Can be None.
+        :type realm: str or None
+        :param user: the user against which policies should be matched. Can be None.
+                     If a user is given, the argument ``realm`` *must* be None because
+                     the ``realm`` attribute of policies is matched against the user realm.
+        :type user: privacyidea.lib.user.User or None
+        :rtype: ``Match``
+        """
+        if scope == SCOPE.ADMIN:
+            raise MatchingError("Match.simple cannot be used for policies with scope ADMIN")
+        if user is None:
+            # If the user is None, we might still have a realm
+            user_object = None
+            username = None
+            resolver = None
+        elif isinstance(user, User):
+            user_object = user
+            if realm is not None:
+                raise MatchingError("Conflicting matching parameters: realm/user")
+            # Username, realm and resolver will be extracted from the user_object parameter
+            username = None
+            realm = None
+            resolver = None
+        else:
+            raise MatchingError("Invalid user")
+        return cls(g, name=None, scope=scope, realm=realm, active=True,
+                   resolver=resolver, user=username, user_object=user_object,
+                   client=g.client_ip, action=action, adminrealm=None, time=None,
+                   sort_by_priority=True)
+
+    @classmethod
+    def admin(cls, g, action, realm):
+        """
+        Matching admin policies with an action and, optionally, a realm.
+        Assumes that the currently logged-in user is an admin, and throws an error otherwise.
+        Policies will be matched against the admin's username and adminrealm,
+        and optionally also the provided realm.
+        :param g: ``flask.g`` object
+        :param action: the policy action
+        :param realm: the realm against which policies should be matched. Can be None.
```

Comment:
Please add in docs, that this is a userrealm

Reply:
done in a11ccd0d1a2a1d94e5c2be6a02fae4454f636279